src/pkg/boltz.ts

```diff
@@ -0,0 +1,467 @@
+    claimPublicKey: Buffer,
+    network: NetworkString,
+    invoiceAmount: number
+  ): Promise<ReverseSwap>;
+  makeClaimTransaction(p: MakeClaimTransactionParams): Transaction;
+  getInvoiceExpireDate(invoice: string): number;
+  getInvoiceValue(invoice: string): number;
+}
+
+interface CreateSwapCommonRequest {
+  type: 'submarine' | 'reversesubmarine';
+  pairId: 'L-BTC/BTC';
+  orderSide: 'buy' | 'sell';
+}
+
+interface CreateSwapCommonResponse {
+  id: string;
+  timeoutBlockHeight: number;
+}
+
+interface SubmarineSwapRequest {
+  invoice: string;
+  refundPublicKey: string;
+}
+
+interface SubmarineSwapResponse {
+  acceptZeroConf: boolean;
+  address: string;
+  bip21: string;
+  blindingKey: string; // private
+  expectedAmount: number;
+  redeemScript: string;
+}
+
+interface ReverseSubmarineSwapRequest {
+  preimageHash: string;
+  invoiceAmount: number;
+  claimPublicKey: string;
+}
+
+interface ReverseSubmarineSwapResponse {
+  blindingKey: string; // private
+  invoice: string;
+  lockupAddress: string;
+  onchainAmount: number;
+  redeemScript: string;
+}
+
+export interface SubmarineSwap {
+  address: string;
+  expectedAmount: number;
+  id: string;
+  redeemScript: string;
+}
+
+export interface ReverseSwap {
+  blindingPrivateKey: string;
+  claimPublicKey: string;
+  id: string;
+  invoice: string;
+  lockupAddress: string;
+  preimage: Buffer;
+  redeemScript: string;
+  timeoutBlockHeight: number;
+}
+
+export interface MakeClaimTransactionParams {
+  utxo: Unspent;
+  claimKeyPair: ECPairInterface;
+  preimage: Buffer;
+  redeemScript: Buffer;
+  destinationScript: Buffer;
+  fee: number;
+  blindingPublicKey: Buffer;
+  timeoutBlockHeight: number;
+}
+
+const feeAmount = 500; // fee for regular liquid tx
+const swapFeeAmount = 500; // fee for Boltz
+// lightning swap invoice amount limit (in satoshis)
+export const DEFAULT_LIGHTNING_LIMITS = { maximal: 4294967, minimal: 50000 };
+export const DEPOSIT_LIGHTNING_LIMITS = {
+  maximal: DEFAULT_LIGHTNING_LIMITS.maximal - feeAmount - swapFeeAmount,
+  minimal: DEFAULT_LIGHTNING_LIMITS.minimal - feeAmount - swapFeeAmount,
+};
+
+export const boltzUrl: Record<NetworkString, string> = {
+  regtest: 'http://localhost:9090',
+  testnet: 'https://testnet.boltz.exchange/api',
+  liquid: 'https://boltz.exchange/api',
+};
+
+export class Boltz implements BoltzInterface {
+  private asset: string;
+  private url: string;
+  private zkp: ZKP;
+
+  constructor(url: string, asset: string, zkp: ZKP) {
+    this.asset = asset;
+    this.url = url;
+    this.zkp = zkp;
+  }
+
+  async getBoltzPair(pair: string): Promise<any> {
+    const data = await this.getApi(`${this.url}/getpairs`);
+    if (!data?.pairs?.[pair]) return;
+    return data.pairs[pair];
+  }
+
+  // return invoice expire date
+  getInvoiceExpireDate(invoice: string): number {
+    const { timeExpireDate } = bolt11.decode(invoice);
+    return timeExpireDate ? timeExpireDate * 1000 : 0; // milliseconds
+  }
```

Comment:
There's a bug on boltz mainnet api, where invoices DON'T include field `timeExpireDate`, resulting in invoices from mainnet to be considered expired.

This changes try to find out the correct expiration date cycling through different strategies and returning by default `Date.now() * 3,600,000`

```suggestion
  getInvoiceExpireDate(invoice: string): number {
    const toMilliseconds = (num: number) => num * 1000;
    const { timeExpireDate, timestamp } = bolt11.decode(invoice);
    if (timeExpireDate) return toMilliseconds(timeExpireDate);
    const { expire_time } = bolt11.decode(invoice).tagsObject;
    const expireTime = toMilliseconds(expire_time ?? 3600);
    const from = timestamp ? toMilliseconds(timestamp) : Date.now();
    return from + expireTime;
  }
```